### DISCUSSION TITLE
How to add a button on user list

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#13655] [Q&A] [answered]
Hi，

I want to add a button called "Simulate Login" to the user list in the admin background.

Used to perform tests that impersonate user identity

I have added the OrchardCore.Users dependency on the Startups of my module

I've copied `UserButtons.cshtml` to the Views folder, but it doesn't react to anything.

![image](https://user-images.githubusercontent.com/15613121/236541684-8e5baad2-310a-436b-be23-c15a47234efc.png)

Is it the dependency order or the razore file name?

**Comment by hyzx86** [comment]
``` c#
    public class MockUserDisplayDriver : DisplayDriver<User>
    {
        public override IDisplayResult Display(User user)
        {
            return Combine(
                Initialize<SummaryAdminUserViewModel>("UserButtons", model => model.User = user).Location("SummaryAdmin", "Actions:2")
            );
        }
    }
```

**Reply by hyzx86** [reply]
but 。。。

The result of this change is that the button is out, but it just renders the existing button twice.

And the button I added doesn't appear


![image](https://user-images.githubusercontent.com/15613121/236546946-fc94b9c0-1cd8-40b5-a34d-d65950cb3ca2.png)


my razor file content is:
``` razor

@model SummaryAdminUserViewModel
@using EasyOC.Authentication;
@using OrchardCore.Entities
@using OrchardCore.Users.Models
@using Microsoft.AspNetCore.Identity
@using OrchardCore.Users
@using EasyOC.Core
@inject UserManager<IUser> UserManager
@inject IAuthorizationService AuthorizationService
@{
    var isCurrentUser = Model.User.UserName == User.Identity.Name;
}

@if (!isCurrentUser && await AuthorizationService.AuthorizeAsync(User, PermissionsProvider.MockUserLoginPermission, Model.User))
{
    <a asp-action="MockLogin" asp-controller="MockUserLogin" asp-route-id="@Model.User.UserId" class="btn btn-secondary btn-sm">@T["MockLogin"]</a>
}

```

**Comment by hyzx86** [comment, marked as answer]
Razor page filename should be `UserButtons.MockUserLogin.cshtml`

and the DisplayDriver 
```
Initialize<SummaryAdminUserViewModel>("UserButtons_MockUserLogin", model => model.User = user).Location("SummaryAdmin", "Actions:2")

```

![TMJ$0L153J~ Q5N)OJYTZ1K](https://user-images.githubusercontent.com/15613121/236592695-7eefbf62-7716-414d-bf75-00bf3c366fd7.png)


![image](https://user-images.githubusercontent.com/15613121/236592913-8457f3d3-9e07-4db0-9992-19efa73ff985.png)

**Comment by hishamco** [comment]
Hope [this](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/main/src/OrchardCoreContrib.Users/README.md) helps you

**Reply by hyzx86** [reply]
Great module, much more polished than I did, thanks for sharing
